pkg/subctl/cmd/root.go

```diff
@@ -163,14 +163,22 @@ func handleNodeLabels(config *rest.Config) error {
 }
 
 func askForGatewayNode(clientset kubernetes.Interface) (struct{ Node string }, error) {
-	// List all nodes and select one
-	allNodes, err := clientset.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: "!node-role.kubernetes.io/master"})
+	// List the worker nodes and select one
+	allNodes, err := clientset.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: "node-role.kubernetes.io/worker"})
 	if err != nil {
 		return struct{ Node string }{}, err
 	}
 	if len(allNodes.Items) == 0 {
-		return struct{ Node string }{}, nil
+		// In some deployments (like KIND), worker nodes are not explicitly labelled. So list non-master nodes.
+		allNodes, err = clientset.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: "!node-role.kubernetes.io/master"})
```

Comment:
```suggestion
		nonMasterNodes, err = clientset.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: "!node-role.kubernetes.io/master"})
```

Reply:
If we use separate variable names for both, then we will have to use additional checks in Line182, Line186.

As you know, allNodes is just a local variable. Instead of having two separate variables, is it okay if I change it from `allNodes` to `workerNodes`?

Reply:
That makes sense @sridhargaddam, is more clear. :+1:

Reply:
Ack. Thanks for reviewing @dfarrell07